File: services/grant-application-service/src/main/java/com/karsunfde/grantsportal/grantapplication/model/Amendment.java
package com.karsunfde.grantsportal.grantapplication.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

/**
 * Amendment to a published grantApplication. FAR 15.206.
 *
 * Amendments are numbered sequentially per grantApplication (0001, 0002, ...).
 * Vendors with proposals-in-progress must acknowledge before deadline.
 *
 * ⚠ Item 9 — {@code changeSummary} is raw HTML, fed into ai-orchestrator
 * for /draft-amendment narrative generation.
 * ⚠ Item 2 — amendment state transitions are audit-log race-prone.
 */
@Document(collection = "amendments")
public class Amendment {

    @Id
    private String id;

    private String grantApplicationId;
    private String agencyId;
    private int number;

    /** ⚠ Item 9 — raw HTML accepted. */
    private String changeSummary;

    private Instant effectiveAt;
    private boolean requiresAcknowledgement;
    private Instant createdAt;

    public Amendment() {}

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getGrantApplicationId() { return grantApplicationId; }
    public void setGrantApplicationId(String grantApplicationId) { this.grantApplicationId = grantApplicationId; }
    public String getAgencyId() { return agencyId; }
    public void setAgencyId(String agencyId) { this.agencyId = agencyId; }
    public int getNumber() { return number; }
    public void setNumber(int number) { this.number = number; }
    public String getChangeSummary() { return changeSummary; }
    public void setChangeSummary(String changeSummary) { this.changeSummary = changeSummary; }
    public Instant getEffectiveAt() { return effectiveAt; }
    public void setEffectiveAt(Instant effectiveAt) { this.effectiveAt = effectiveAt; }
    public boolean isRequiresAcknowledgement() { return requiresAcknowledgement; }
    public void setRequiresAcknowledgement(boolean requiresAcknowledgement) { this.requiresAcknowledgement = requiresAcknowledgement; }
    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }
}

File: frontend/src/app/models/amendment.ts
/**
 * Amendment to a published grantApplication (FAR 15.206).
 *
 * Numbered sequentially per grantApplication. Issuance is restricted to CO.
 * Vendors with proposals-in-progress must acknowledge before deadline;
 * acknowledgement state is tracked here.
 */
export interface Amendment {
  id: string;
  grantApplicationId: string;
  number: number;                  // 0001, 0002, ...
  changeSummary: string;
  effectiveAt: string;             // ISO
  requiresAcknowledgement: boolean;
  acknowledgedBy: string[];        // vendor IDs
  issuedBy: string;                // CO user id
  issuedAt: string;                // ISO
}

export interface AmendmentCreate {
  changeSummary: string;
  effectiveAt: string;
  requiresAcknowledgement: boolean;
}
